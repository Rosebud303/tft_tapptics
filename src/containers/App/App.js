import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import MainPage from '../MainPage';
import Champions from '../Champions/Champions';
import Items from '../Items/Items';
import TeamBuilder from '../TeamBuilder/TeamBuilder';
import Synergies from '../Synergies/Synergies';
import NoMatch from '../../components/NoMatch/NoMatch';

class App extends Component {
  render() {
    return (
    <div className="App">
      <header className="App-header">
        <h1>TFTapptics</h1>
        <MainPage />
      </header>
      <Switch>
        <Route exact path="/" component={MainPage} />
        <Route exact path="/team-builder" component={TeamBuilder} />
        <Route exact path="/items" component={Items} />
        <Route exact path="/champions" component={Champions} />
        <Route exact path="/synergies" component={Synergies} />
        <Route component={NoMatch} />
      </Switch>
    </div>
    )
  }
}

export default App;
