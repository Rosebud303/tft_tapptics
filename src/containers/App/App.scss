body {
  box-sizing: border-box;
  font-family: $body-font;
  margin: 0;
  height: 100vh;
}