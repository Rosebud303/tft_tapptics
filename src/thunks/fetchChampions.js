import { fetchData } from '../api/fetchData';
import { isLoading, setError, setChampions } from '../actions';

export const fetchChampions = url => {
  return async dispatch => {
    try {
      dispatch(isLoading(true))
      const data = await fetchData(url);
      dispatch(setChampions(data))
      dispatch(isLoading(false))
    } catch (error) {
      dispatch(setError(error.message))
    }
  }
}