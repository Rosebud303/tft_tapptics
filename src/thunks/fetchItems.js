import { fetchData } from '../api/fetchData';
import { isLoading, setError, setItems } from '../actions';

export const fetchItems = url => {
  return async dispatch => {
    try {
      dispatch(isLoading(true));
      const data = fetchData(url);
      dispatch(setItems(data.data));
      dispatch(isLoading(false))
    } catch (error) {
      dispatch(setError(error.message))
    }
  } 
}