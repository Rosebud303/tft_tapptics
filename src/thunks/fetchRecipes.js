import { fetchData } from '../api/fetchData';
import { isLoading, setError, setRecipes } from '../actions';

export const fetchRecipes = url => {
  return async dispatch => {
    try {
      dispatch(isLoading(true));
      const data = await fetchData(url);
      console.log(data.data)
      dispatch(setRecipes(data.data));
      dispatch(isLoading(false))
    } catch (error) {
      dispatch(setError(error.message))
    }
  } 
}