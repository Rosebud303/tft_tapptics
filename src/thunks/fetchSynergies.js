import { fetchData } from '../api/fetchData';
import { isLoading, setError, setSynergies } from '../actions';

export const fetchSynergies = url => {
  return async dispatch => {
    try {
      dispatch(isLoading(true));
      const data = await fetchData(url);
      dispatch(setSynergies(data.data));
      dispatch(isLoading(false));
    } catch (error) {
      dispatch(setError(error.message));
    }
  }
}