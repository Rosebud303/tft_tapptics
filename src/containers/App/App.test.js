import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { fetchChampions } from '../../thunks/fetchChampions';
import { fetchSynergies } from '../../thunks/fetchSynergies';
import { fetchItems } from '../../thunks/fetchSynergies';

jest.mock('../../thunks/fetchChampions');
jest.mock('../../thunks/fetchSynergies');
jest.mock('../../thunks/fetchItems');


  let mockChamps = [{
    "id": "1",
    "type": "champions",
    "attributes": {
        "data": {
            "id": 1,
            "name": "Aatrox, the Darkin Blade",
            "champion_thumbnail": "https://ddragon.leagueoflegends.com/cdn/9.13.1/img/champion/Aatrox.png",
            "cost": 3,
            "health": [
                650,
                1170,
                2340
            ],
            "dmg": 65,
            "armor": 25,
            "mr": 20,
            "atk_spd": 0.65,
            "range": "■□□□",
            "ability_thumbnail": "https://raw.communitydragon.org/latest/game/assets/characters/aatrox/hud/icons2d/aatrox_q3.png",
            "ability_info": {
                "title": "The Darkin Blade",
                "attributes": [
                    {
                        "damage": [
                            300,
                            500,
                            700
                        ]
                    }
                ],
                "descrption": "Aatrox cleaves the area in front of him, dealing damage to enemies inside it"
            },
            "model_img": null,
            "created_at": "2019-07-18T23:35:03.507Z",
            "updated_at": "2019-07-18T23:35:03.507Z"
        },
        "origin_class_type": {
            "data": [
                {
                    "id": "2",
                    "type": "origin_class_type",
                    "attributes": {
                        "data": {
                            "id": 2,
                            "name": "Blademaster",
                            "thumbnail": "https://img.rankedboost.com/wp-content/plugins/league/assets/tft/Blademaster.png",
                            "summary": "Blademasters have a chance to strike additional times each attack.",
                            "tier_info": [
                                " (2)  Blademasters 35% chance to strike 1 additional attack",
                                " (4)  Blademasters 35% chance to strike 2 additional"
                            ],
                            "tiers": [
                                2,
                                4
                            ],
                            "created_at": "2019-07-18T23:35:03.305Z",
                            "updated_at": "2019-07-18T23:35:03.305Z"
                        }
                    }
                },
                {
                    "id": "11",
                    "type": "origin_class_type",
                    "attributes": {
                        "data": {
                            "id": 11,
                            "name": "Demon",
                            "thumbnail": "https://img.rankedboost.com/wp-content/plugins/league/assets/tft/Demon.png",
                            "summary": "Attacks from Demons have a chance on hit to burn all of an enemy's mana and deal that much as true damage.",
                            "tier_info": [
                                " (2)  Demons have a 40% Chance on Hit to Mana Burn",
                                " (4)  Demons have a 60% Chance on Hit to Mana Burn",
                                " (6)  Demons have a 80% Chance on Hit to Mana Burn"
                            ],
                            "tiers": [
                                2,
                                4,
                                6
                            ],
                            "created_at": "2019-07-18T23:35:03.379Z",
                            "updated_at": "2019-07-18T23:35:03.379Z"
                        }
                    }
                }
            ]
        }
    }
}]

describe('App', () => {
  
  describe('Container', () => {
    let mockFunc;
    let mockFetchSynergies;
    let mockFetchItems;
    let mockFetchRecipes;
    let wrapper;

    beforeEach(() => {
      mockFunc = jest.fn()
      mockFetchSynergies = jest.fn();
      mockFetchItems = jest.fn();
      mockFetchRecipes = jest.fn();
      wrapper = shallow(<App champions={mockChamps}
                             fetchChampions={mockFunc}
                             fetchSynergies={mockFetchSynergies}
                             fetchItems={mockFetchItems}
                             fetchRecipes={mockFetchRecipes}
                             />);
    });


    it('should match the snapshot with all the data passed in', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapDispatchToProps', () => {
    let fetchItems = jest.fn();
    let mockDispatch = jest.fn();
    
    it('should call fetchChampions with the correct params', () => {
      let mockUrl = 'www.TFT.com';
      let mockDispatch = jest.fn();
      let actionsToDispatch = fetchChampions(mockUrl);
      let mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.fetchChampions(actionsToDispatch);
      expect(mockDispatch).toHaveBeenCalledWith(actionsToDispatch);
    });

    it('should call fetchItems with the correct params', () => {
        const mockState = {
            items: [{name: 'Golden Spatula'}]
          }

          const actionToDispatch = fetchItems(mockState);
          const mappedProps = mapDispatchToProps(mockDispatch);
          mappedProps.fetchItems(mockState);
          
          expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
      });

      it('should call fetchSynergies with the correct params', () => {
        const mockState = {
            items: [{name: 'Golden Spatula'}]
          }
        const actionToDispatch = fetchSynergies(mockState);
        const mappedProps = mapDispatchToProps(mockDispatch);
        mappedProps.fetchSynergies(mockState);
        
        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
      });
  }); 
});