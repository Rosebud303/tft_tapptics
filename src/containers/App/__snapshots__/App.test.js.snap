// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all the data passed in 1`] = `
<div
  className="App"
>
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/items"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/champions"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/synergies"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/details"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/carousel"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
